sklearn/tests/test_pipeline.py

```diff
@@ -1029,3 +1031,41 @@ def test_make_pipeline_memory():
     assert_true(pipeline.memory is None)
 
     shutil.rmtree(cachedir)
+
+
+@pytest.mark.parametrize(['est', 'pattern'], [
+    (Pipeline([('transf', Transf()), ('clf', FitParamT())]),
+     r'''\[Pipeline\].*\(step 1 of 2\) Fitting transf.* elapsed=.*\n'''
+     r'''\[Pipeline\].*\(step 2 of 2\) Fitting clf.* elapsed=.*\n$'''),
+    (Pipeline([('transf', Transf()), ('clf', None)]),
+     r'''\[Pipeline\].*\(step 1 of 1\) Fitting transf.* elapsed=.*\n$'''),
+    (Pipeline([('transf', None), ('mult', Mult())]),
+     r'''\[Pipeline\].*\(step 1 of 1\) Fitting mult.* elapsed=.*\n$'''),
+    (FeatureUnion([('mult1', Mult()), ('mult2', Mult())]),
+     r'''\[FeatureUnion\].*\(step 1 of 2\) Fitting mult1.* elapsed=.*\n'''
+     r'''\[FeatureUnion\].*\(step 2 of 2\) Fitting mult2.* elapsed=.*\n$'''),
+    (FeatureUnion([('mult1', None), ('mult2', Mult()), ('mult3', None)]),
+     r'''\[FeatureUnion\].*\(step 1 of 1\) Fitting mult2.* elapsed=.*\n$'''),
+])
+@pytest.mark.parametrize('method', ['fit', 'fit_transform', 'fit_predict'])
+def test_verbose(est, method, pattern, capsys):
+    try:
+        func = getattr(est, method)
+    except AttributeError:
+        return
```

Comment:
Rather than silently skipping tests , it would be better to generate an list of combinations that is actually valid.

```py
parameter_grid = ((est, pattern, method)
                  for (est, pattern), method in itertools.product(
                      [
                       # possibly define the following outside of this loop
                       (Pipeline([('transf', Transf()), ('clf', FitParamT())]),
                        [...]
                      ],
                      ['fit', 'fit_transform', 'fit_predict'])
                  if hasattr(est, method) and ...)

@pytest.mark.parametrize('est, pattern, method', parameter_grid):
```